src/store/participantsStore.js

```diff
@@ -788,28 +792,31 @@ const actions = {
 			return
 		}
 
-		commit('setInCall', {
-			token,
-			sessionId: participantIdentifier.sessionId,
-			flags,
+		// Preparing the event listener for the signaling-join-call event
+		EventBus.once('signaling-join-call', () => {
+			commit('setInCall', {
+				token,
+				sessionId: participantIdentifier.sessionId,
+				flags,
+			})
+			commit('finishedConnecting', { token, sessionId: participantIdentifier.sessionId })
 		})
 
-		const actualFlags = await joinCall(token, flags, silent, recordingConsent)
-
-		const updatedData = {
-			inCall: actualFlags,
-		}
-		commit('updateParticipant', { token, attendeeId: attendee.attendeeId, updatedData })
-
-		EventBus.once('signaling-users-in-room', () => {
+		// Preparing the event listener for the signaling-join-call-failed event
+		EventBus.once('signaling-join-call-failed', () => {
 			commit('finishedConnecting', { token, sessionId: participantIdentifier.sessionId })
 		})
 
-		setTimeout(() => {
-			// If by accident we never receive a users list, just switch to
-			// "Waiting for others to join the call …" after some seconds.
-			commit('finishedConnecting', { token, sessionId: participantIdentifier.sessionId })
-		}, 10000)
```

Comment:
It's guaranteed to receive one of the above signaling events as we are the base server

Reply:
Note that `finishedConnecting` does not mean `It was possible to join the call`, but `After joining the call, the list of participants in the call was received`. Therefore this change breaks its behaviour.

After joining a call without HPB _Connecting_ was shown in the call view until the list of participants in the call was received. If there were no other participants in the call _Connecting_ was replaced by _Waiting for others to join the call_, and if there were other participants then a connection was established with them. Now, if there are other participants, _Waiting for others to join the call_ will be briefly shown until the participant list is received, which is not correct.

If there is HPB, however, the _Connecting_ state was wrong, as `usersInRoom` is never triggered in that case (probably `usersChanged` should have been used instead) :shrug: So _Connecting_ was shown for ten seconds when there were no other participants in the call.

Reply:
Fixed, yet one issue is that I couldn't find a signaling event on `base` level for receiving participants list. 
`usersInRoom` works for internal signaling
However, there was no event triggered in external signaling. Am I missing one or I can just create a new one?

Reply:
You could listen to [`usersChanged`](https://github.com/nextcloud/spreed/blob/157e1751be3e571c961e0e28d1141f1b33473c18/src/utils/signaling.js#L1500). This is the counterpart of `usersInRoom` for the external signaling server, and it is in fact [the one used to check which participants are in the call](https://github.com/nextcloud/spreed/blob/32ba79dec837d3c2f33015ccb2522bc4b4534b46/src/utils/webrtc/webrtc.js#L521) to connect to them (similarly to [what is done with `usersInRoom`](https://github.com/nextcloud/spreed/blob/32ba79dec837d3c2f33015ccb2522bc4b4534b46/src/utils/webrtc/webrtc.js#L563)).

A little detail to keep in mind. While it would be really unusual, it is technically possible that a previous participant update is received after joining the call and before receiving the updated list of participants that includes you. So while in general it would be enough to wait for the first `usersInRoom`/`usersChanged` event received after joining the call the best would be to wait for the first that includes your own session with [an `inCall` flag different than disconnected](https://github.com/nextcloud/spreed/blob/9c30404613fb2694e2a1c60bb2da1f572f332874/src/constants.js#L133).

But keep in mind that when the external signaling server is used there are two different session IDs, the Nextcloud session ID (probably the one that you should check) and the signaling session ID. The `participants->update` message sent by the external signaling server (which becomes the `usersChanged` event) should include both, while the `usersInRoom` message will only have `sessionId` (because in that case there is just one).

Reply:
From testing, I spotted that **updated** `usersChanged` comes before `joinCall` :')

Reply:
That is [expected](https://github.com/nextcloud/spreed/blob/32ba79dec837d3c2f33015ccb2522bc4b4534b46/src/utils/webrtc/webrtc.js#L575-L581) (unfortunate, but expected :-P ). The join call request sent to the Nextcloud server causes the signaling server to send a signaling message with the updated participant list. However, as the signaling message is sent through the WebSocket in some cases it can be received (which causes [the `usersChanged` event to be emitted](https://github.com/nextcloud/spreed/blob/24d2865eb3fda780e5f36439ad1f2497c42d4f59/src/utils/signaling.js#L1500)) before the HTTP request response for the join call request is received (which causes [the `joinCall` event to be emitted](https://github.com/nextcloud/spreed/blob/24d2865eb3fda780e5f36439ad1f2497c42d4f59/src/utils/signaling.js#L277)).

What you could do is, rather than finish connecting once the updated participant list is received, finish connecting once the updated participant list is received if `joinCall` was already received, and finish connecting once `joinCall` is received if the updated participant list was already received between starting to join the call and actually joining the call.

Reply:
![image](https://github.com/user-attachments/assets/5d587fb8-e5fc-4ac0-8f50-790d241ac88f)